third_party/libusb/webport/src/libusb-to-webusb-adaptor.js

```diff
@@ -88,14 +111,50 @@ GSC.LibusbToWebusbAdaptor = class extends GSC.LibusbToJsApiAdaptor {
 
   /** @override */
   async openDeviceHandle(deviceId) {
-    // TODO(#429): Implement this method.
-    throw new Error('Not implemented');
+    const deviceState = this.getDeviceByIdOrThrow_(deviceId);
+    // WebUSB doesn't have a concept of device handles, so generate a handle
+    // from a counter.
+    if (deviceState.handles.size === 0) {
+      // Ask WebUSB to open the device, but only for the first opened handle.
+      // Avoid concurrent WebUSB open()/close() calls, since WebUSB throws on
+      // these, hence track the ongoing request's promise in the device state
+      // and wait for it.
+      if (!deviceState.openOperationPromise) {
+        // Note: no "await" here, since we need to set `openOperationPromise`
+        // synchronously and therefore let all subsequent requests wait on it.
+        deviceState.openOperationPromise = openWebusbDevice(deviceState);
+      }
+      await deviceState.openOperationPromise;
+      deviceState.openOperationPromise = null;
```

Comment:
It's a question whether this should be wrapped inside "finally" section, so that one open() failure doesn't cause all subsequent ones to fail as well (especially those that are started much later).

Reply:
(did that)